src/verify_expr.ml

```diff
@@ -1774,11 +1788,14 @@ module VerifyExpr(VerifyProgramArgs: VERIFY_PROGRAM_ARGS) = struct
                   | _ -> get_unique_var_symb_ "value" t (gh = Ghost)
                 end
                 end
-              | Expr e -> eval_h h env e (fun h env v -> cont h env (Some v))
-              | Term t -> cont h env (Some t)
-              | Unspecified -> cont h env (match gh with Ghost -> Some (get_unique_var_symb_ "value" t true) | Real -> None)
-            end $. fun h env value ->
-            assume_field h env sn tparams f t0 targs gh addr RegularPointsTo value coef $. fun h ->
+              | Expr e -> eval_h h env e (fun h env v -> cont h env t0 (Some v))
+              | Term t -> cont h env t0 (Some t)
+              | MaybeUninitTerm t -> 
+                let t = prover_convert_term t t0 (instantiate_type tpenv t0) in
+                cont h env (option_type t0) (Some t)
+              | Unspecified -> cont h env t0 (match gh with Ghost -> Some (get_unique_var_symb_ "value" t true) | Real -> None)
+            end $. fun h env t0 value ->
+            assume_field h env sn tparams f t0 targs gh addr pointsto_kind value coef $. fun h ->

```

Comment:
It seems wrong to pass the option type as the field type? assume_field is aware that if the pointsto_kind is MaybeUninit, then the value is an option of the field type.

Reply:
Except that there does seem to be a bug in assume_field: it will assume_bounds of tv at frange even if pointsto_kind is MaybeUninit. But I think it's better to fix that bug than to pass the option type as the field type. If pointsto_kind is MaybeUninit and tv is Some tv, then assume_field should assume_bounds tv (option_type frange). (Currently assume_bounds does nothing for option types, but we might want to have it assume the bounds of the value inside in the future.)

Reply:
I changed the `assume_bounds` behaviour as suggested and the option field type is no longer passed.

Reply:
It seems as if you *are* still passing the option field type?

Reply:
Forgot to push that change, my apologies!